flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/types/ResourceProfile.java

```diff
@@ -292,7 +349,15 @@ public boolean isMatching(ResourceProfile required) {
 
 	@Override
 	public int compareTo(@Nonnull ResourceProfile other) {
-		int cmp = Integer.compare(this.getMemoryInMB(), other.getMemoryInMB());
+		if (this == other) {
+			return 0;
+		} else if (this == UNKNOWN) {
```

Comment:
Are there cases where we have to compare with `UNKNOWN`?

Reply:
It's possible. E.g., we may need to sort slot requests according to their resources, of which some might be unknown and others might be specified.
I think `ResourceProfile#compareTo` should not assume whether an unknown profile will be compared or not. It should be able to handle call possible values as a `Comparable`.